sklearnex/linear_model/tests/test_linear.py

```diff
@@ -80,3 +85,35 @@ def test_sklearnex_import_elastic():
     assert "daal4py" in elasticnet.__module__
     assert_allclose(elasticnet.intercept_, 1.451, atol=1e-3)
     assert_allclose(elasticnet.coef_, [18.838, 64.559], atol=1e-3)
+
+
+@pytest.mark.parametrize("dataframe,queue", get_dataframes_and_queues())
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_sklearnex_reconstruct_model(dataframe, queue, dtype):
+    from sklearnex._device_offload import _transfer_to_host
+    from sklearnex.linear_model import LinearRegression
+
+    seed = 42
+    num_samples = 3500
+    num_features, num_targets = 14, 9
+
+    gen = np.random.default_rng(seed)
+    intercept = gen.random(size=num_targets, dtype=dtype)
+    coef = gen.random(size=(num_targets, num_features), dtype=dtype).T
+
+    X = gen.random(size=(num_samples, num_features), dtype=dtype)
+    gtr = X @ coef + intercept[np.newaxis, :]
+
+    X = _convert_to_dataframe(X, sycl_queue=queue, target_df=dataframe)
+
+    linreg = LinearRegression(fit_intercept=True)
+    linreg.coef_ = coef.T
+    linreg.intercept_ = intercept
+
+    y_pred = linreg.predict(X)
+
+    # TODO Implement more clear way to transfer dpctl/dpnp arrays to host
+    _, hostdata = _transfer_to_host(queue, y_pred)

```

Comment:
Other tests that use _convert_to_dataframe() are using `_as_numpy` to assert_allclose - maybe this would be the more clear way to transfer